framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/deployment/KieServerOperatorDeployment.java

```diff
@@ -58,7 +59,10 @@ public void scale(int instances) {
     public void waitForScale() {
         KieApp kieApp = kieAppClient.withName(OpenShiftConstants.getKieApplicationName()).get();
 
-        Integer replicas = getAssociatedServerObject(kieApp.getStatus().getApplied()).getReplicas();
+        Integer replicas = Optional.ofNullable(kieApp.getStatus())
+                                   .map(status -> status.getApplied())
+                                   .map(applied -> getAssociatedServerObject(applied).getReplicas())
+                                   .orElse(null);
         if (replicas == null) {
             // If replicas are not set in custom resource then get them from deployment config.
             replicas = getOpenShift().getDeploymentConfig(getServiceName()).getSpec().getReplicas();
```

Comment:
As we are using Optional here, can be code from if blocked moved to orElse part? So we would not need to return null to Integer replicas and then check it again for null.

Reply:
Good point! Already changed in all the operator deployments.